subxt/src/backend/mod.rs

```diff
@@ -325,9 +325,417 @@ pub enum TransactionStatus<Hash> {
+            let response = backend
+                .storage_fetch_values(
+                    ["ID1".into(), "ID2".into(), "ID3".into()].into(),
+                    crate::utils::H256::random(),
+                )
+                .await
+                .unwrap();
+
+            let response = response
+                .map(|x| x.unwrap())
+                .collect::<Vec<StorageResponse>>()
+                .await;
+
+            let expected = vec![
+                storage_response("ID1", "Data1"),
+                storage_response("ID2", "Data2"),
+                storage_response("ID3", "Data3"),
+            ];
+
+            assert_eq!(expected, response)
+        }
+
+        #[tokio::test]
+        async fn storage_fetch_value() {
+            // Setup
+            let mock_data = [
+                (
+                    "ID1",
+                    Err(RpcError::DisconnectedWillReconnect(
+                        "Reconnecting".to_string(),
+                    )),
+                ),
+                ("ID1", bytes("Data1")),
+            ];
+            let rpc_client = build_mock_client(mock_data, vec![]).await;
+
+            // Test
+            let backend: LegacyBackend<Conf> = LegacyBackend::builder().build(rpc_client);
+            let response = backend
+                .storage_fetch_value("ID1".into(), crate::utils::H256::random())
+                .await
+                .unwrap();
+
+            let response = response.unwrap();
+            assert_eq!("Data1".to_owned(), String::from_utf8(response).unwrap())
+        }
+
+        #[tokio::test]
+        /// This test should cover following methods:
+        /// - `genesis_hash`
+        /// - `block_header`
+        /// - `block_body`
+        /// - `latest_finalized_block`
+        /// - `current_runtime_version`
+        /// - `current_runtime_version`
+        /// - `call`
+        /// The test covers them because they follow the simple pattern of:
+        /// ```no_run
+        ///  async fn THE_THING(&self) -> Result<T::Hash, Error> {
+        ///    retry(|| <DO THE THING> ).await
+        ///  }
+        /// ```
+        async fn simple_fetch() {
+            let hash = crate::utils::H256::random();
+
+            // Setup
+            let mock_data = vec![
+                (
+                    "chain_getBlockHash".into(),
+                    Err(RpcError::DisconnectedWillReconnect(
+                        "Reconnecting".to_string(),
+                    )),
+                ),
+                ("chain_getBlockHash".into(), Ok(Some(hash))),
+            ];
+            let rpc_client = build_mock_client(mock_data, vec![]).await;
+
+            // Test
+            let backend: LegacyBackend<Conf> = LegacyBackend::builder().build(rpc_client);
+            let response = backend.genesis_hash().await.unwrap();
+
+            assert_eq!(hash, response)
+        }
+
+        #[tokio::test]
+        /// This test should cover following methods:
```

Comment:
Not sure whether we should be extra verbose say: ` This test should cover the logic in the following methods`

Your example below is explains it very well

Reply:
done